src/skimage/registration/_affine.py

```diff
@@ -0,0 +1,551 @@
+    ----------
+    shape : tuple of (int, ...)
+        Shape of the array.
+    matrix : ndarray
+        Homogeneous matrix.
+
+    Returns
+    -------
+    error : ndarray
+        Norm of the displacement given by the transform.
+    """
+    # Create a regular set of points on the grid
+    slc = [slice(0, n) for n in shape]
+    N = np.prod(shape)
+    points = np.stack([*[x.ravel() for x in np.mgrid[slc]], np.ones(N)])
+    # compute the displacement
+    delta = matrix @ points - points
+    error = np.linalg.norm(delta[: len(shape)], axis=0).reshape(shape)
+    return error
+
+
+def _parameter_vector_to_matrix(parameters, model, ndim):
+    """
+    Transform a vector of parameter to an affine matrix.
+
+    Parameters
+    ----------
+    parameters : ndarray
+        Vector of parameters (see note).
+    model : {'affine', 'euclidean', 'translation'}
+        Motion model 'affine', 'euclidean' or 'translation'.
+    ndim : int
+        Image dimension (any dimensions).
+
+    Returns
+    -------
+    matrix : ndarray
+        Homogeneous matrix
+
+    Raises
+    ------
+    NotImplementedError
+        For unsupported motion models.
+
+    Note
+    ----
+    The parameters are:
+    - translation : [dy,dx] in 2D or [dz,dy,dx] in 3D,
+    - euclidean : [dy,dx,theta] in 2D or [dz,dy,dx,alpha,beta,gamma] in 3D,
+    - affine: the top of the homogeneous matrix minus identity.
+    """
+
+    # Test if the parameter is actually a homogeneous matrix already
+    if parameters.shape == (ndim + 1, ndim + 1):
+        return parameters
+
+    if model.lower() == "translation":
+        matrix = np.eye(ndim + 1)
+        matrix[:ndim, -1] = parameters.ravel()
+    elif model.lower() == "euclidean":
+        matrix = np.eye(ndim + 1)
+        # Rotations for each planes
+        for k, a in enumerate(combinations(range(ndim), 2)):
+            c, s = cos(parameters[ndim + k]), sin(parameters[ndim + k])
+            rot = np.eye(ndim + 1)
+            rot[a[0], a[0]] = c
+            rot[a[0], a[1]] = -s
+            rot[a[1], a[1]] = c
+            rot[a[1], a[0]] = s
+            matrix = matrix @ rot
+        # Translation along each axis
+        trans = np.eye(ndim + 1)
+        trans[:ndim, -1] = parameters[:ndim].ravel()
+        matrix = matrix @ trans
+    elif model.lower() == "affine":
+        matrix = np.eye(ndim + 1, dtype=parameters.dtype)
+        matrix[:ndim, -1] = parameters[:ndim].ravel()
+        matrix[:ndim, :ndim] = parameters[ndim:].reshape(ndim, ndim) + np.eye(ndim)
+    else:
+        raise NotImplementedError(f"Model {model} is not supported")
+    return matrix
+
+
+def _scale_parameters(parameters, model, ndim, scale):
+    """
+    Scale the parameter vector or homogeneous matrix.
+
+    Parameters
+    ----------
+    parameters : ndarray
+        Vector of parameters or homogeneous matrix.
+    model : {'affine', 'euclidean', 'translation'}
+        Motion model 'affine', 'euclidean' or 'translation'.
```

Comment:
```suggestion
        Motion model 'affine', 'euclidean', or 'translation'.
```
Although it's not strict, we try to use the Oxford comma by convention.